### ISSUE TITLE
RDP error connection - Due to CredSSP encryption

### ISSUE THREAD

**Issue by capl-corp** [mRemoteNG/mRemoteNG#978]
Can't connect trough RDP 

## Expected Behavior
Whe I tried to connect by RDP to windows servers receive the message: "An authentication error has occurred... This could be due to CredSSP encryption oracle remediation.

## Current Behavior
![image](https://user-images.githubusercontent.com/39173634/39897067-6008687a-5476-11e8-884e-451a1a2ade5d.png)

## Possible Solution
The only solution i found is ussing the last version of remote desktop from MS store

## Steps to Reproduce (for bugs)
This happend with every version of windows server (2008, 2008r2, 2012)

## Your Environment

Microsoft Windows [Version 10.0.17134.48]

**Comment by cjwxxx** [comment]
I encountered this issue on a set of servers recently, I believe is related to whether the RDP server and/or client have the Windows update discussed in 'CredSSP updates for CVE-2018-0886'  https://go.microsoft.com/fwlink/?linkid=866660

There is discussion and workarounds here:
https://serverfault.com/questions/911590/this-could-be-due-to-credssp-encryption-oracle-remediation-rdp-to-windows-10-p

I have no control of patch levels of the servers I connect to, so temporarily applied the 'insecure' fix i.e. set local group policy 'Encryption Oracle Remediation' to 'Vulnerable'. After this the problem RDP connections were working again.

**Comment by chrvip** [comment]
@cjwxx but it will be not safe. The better way is to use mstsc before new version

**Comment by kmscode** [comment]
Per the MS doc, patched clients cannot connect to unpatched servers by default. 

Again, mRemoteNG uses MS provided classes to make remote desktop connections. We have no control when they decide/need to change functionality like this.

I've updated the wiki with full details here:
https://github.com/mRemoteNG/mRemoteNG/wiki/Prerequisites#credssp-updates-for-cve-2018-0886

Options are to:

- Patch the servers
- Set the policy to vulnerable
- Uninstall KB4103727

Not fun... But could have been worse, could have walked unto this problem on a Monday morning.

@cjwxxx is correct regarding the solution/workaround, @chrvip is also correct about the fact that this leaves you vulnerable. However, this is the corner that MS has painted us into.

Unfortunately there's nothing else for us to do from the mRemoteNG side. Closing this issue accordingly.
